src/Stripe.net/Services/_base/Service.cs

```diff
@@ -252,8 +252,23 @@ protected TEntityReturned CreateEntity(BaseOptions options, RequestOptions reque
                 options,
                 requestOptions);
 
+            options = options ?? new ListOptions();
+            bool iterateBackward = false;
+
+            // Backward iterating activates if we have an `EndingBefore`
+            // constraint and not a `StartingAfter` constraint
+            if (!string.IsNullOrEmpty(options.EndingBefore) && string.IsNullOrEmpty(options.StartingAfter))
+            {
+                iterateBackward = true;
+            }
+
             while (true)
             {
+                if (iterateBackward)
+                {
+                    page.Reverse();
```

Comment:
I'm not sure why you have to reverse the whole page instead of plucking from the first element in the case of `EndingBefore`?

Reply:
I reversed because this would make the items in order, and it's also what we're doing in Ruby. 

Because we're iterating backwards, the actual page comes in in order.

Without reversing the page, assuming ID's are incrementing. Paging with the Starting After (with pages of size 3) would look like:

10
9
8
7
6
5
4
3
2
1

And paging with EndingBefore:

3
2
1
4
5
6
7
8
9
10

By reversing the underlying list, when you use Ending before, you get
1
2
3
4
5
6
7
8
9
10